provider/pkg/gen/go-templates/yaml/configGroup.tmpl

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// *** WARNING: this file was generated by pulumigen. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package yaml
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+// ConfigGroup creates a set of Kubernetes resources from Kubernetes YAML.
+type ConfigGroup struct {
+	pulumi.ResourceState
+
+	Resources map[string]pulumi.Resource
+}
+
+// The set of arguments for constructing a ConfigGroup resource.
+type ConfigGroupArgs struct {
+	// Files is a set of paths, globs, or URLs that uniquely identify files.
+	Files []string
+	// YAML is list of strings containing Kubernetes resource definitions in YAML.
+	YAML []string
+	// Objs is a collection of object maps representing Kubernetes resources.
+	Objs []map[string]interface{}
+	// Transformations is an optional list of transformations to apply to Kubernetes resource definitions
+	// before registering with the engine.
+	Transformations []Transformation
+	// ResourcePrefix is an optional prefix for the auto-generated resource names. For example, a resource named `bar`
+	// created with resource prefix of `"foo"` would produce a resource named `"foo-bar"`.
+	ResourcePrefix string
+}
+
+// NewConfigGroup registers a new resource with the given unique name, arguments, and options.
+func NewConfigGroup(ctx *pulumi.Context,
+	name string, args *ConfigGroupArgs, opts ...pulumi.ResourceOption) (*ConfigGroup, error) {
+
+	// Register the resulting resource state.
+	configGroup := &ConfigGroup{
+		Resources: map[string]pulumi.Resource{},
+	}
+	err := ctx.RegisterComponentResource("kubernetes:yaml:ConfigGroup", name, configGroup, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Now provision all child resources by parsing the YAML files.
+	if args != nil {
+		// Honor the resource name prefix if specified.
+		if args.ResourcePrefix != "" {
+			name = args.ResourcePrefix + "-" + name
+		}
+
+		// Parse and decode the YAML files.
+		rs, err := parseDecodeYamlFiles(ctx, args, true, pulumi.Parent(configGroup))
+		if err != nil {
+			return nil, err
+		}
+		configGroup.Resources = rs
+
+		// Finally, register all of the resources found.
+		err = ctx.RegisterResourceOutputs(configGroup, pulumi.Map{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "registering child resources")
+		}
```

Comment:
Should this be outside the `if` so that we always finish the resource registration?

(also for ConfigFile)